lib/clients/AwsClient.py

```diff
@@ -11,11 +11,15 @@ def __init__(self, operation_name, configuration, directory_persistent, director
                  poll_maximum_time):
         super(AwsClient, self).__init__(operation_name, configuration, directory_persistent, directory_work_list,
                                         poll_delay_time, poll_maximum_time)
-        self.__awsCredentials = {
-            'access_key_id': configuration['access_key_id'],
-            'secret_access_key': configuration['secret_access_key'],
-            'region_name': configuration['region_name']
-        }
+        if not 'credhub_url' in configuration:
+            self.__setCredentials(
+                configuration['access_key_id'], configuration['secret_access_key'], configuration['region_name'])
+        else:
+            self.logger.info('fetching creds from credhub')
+            credentials = self._get_credentials_from_credhub(configuration)
```

Comment:
What happens if there is an error in getting these credentials? Ideally, creation of client should fail and backup should be failed. Do we need to raise exception explicitly like in case [AwsClient.py#L37](https://github.com/cloudfoundry-incubator/service-fabrik-backup-restore/blob/a532408a5cdce672eb6dc42fc065aa388a7784b3/lib/clients/AwsClient.py#L37)

Reply:
@ketakig12 But wouldnt it anyways lead to a run time error and it bubling up & leading to backup failure? Is there a mechanism to handle run time errors centrally?

Reply:
I guess it should be handled at [index.py#L33](https://github.com/cloudfoundry-incubator/service-fabrik-backup-restore/blob/master/lib/clients/index.py#L33), but please test the same.

Reply:
Yes. Tested.